internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -267,15 +274,27 @@ func (r *delegatingResolver) updateProxyResolverState(state resolver.State) erro
 	err := r.updateClientConnStateLocked()
 	// Another possible approach was to block until updates are received from
 	// both resolvers. But this is not used because calling `New()` triggers
-	// `Build()`  for the first resolver, which calls `UpdateState()`. And the
+	// `Build()` for the first resolver, which calls `UpdateState()`. And the
 	// second resolver hasn't sent an update yet, so it would cause `New()` to
 	// block indefinitely.
 	if err != nil {
-		r.targetResolver.ResolveNow(resolver.ResolveNowOptions{})
+		go func() {
+			r.childMu.Lock()
+			defer r.childMu.Unlock()
+			if !r.isClosedLocked() {
```

Comment:
Probably a nit, but IMO it is simpler to delete the helper and here just check if `r.targetResolver != nil` since that's what really matters.

Reply:
Deleted the helper and added a check.